core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -138,21 +138,25 @@ object ConsoleConsumer extends Logging {
             throw e
           }
       }
-      checkErr(formatter)
+      if (checkErr(output, formatter)) {
+        // Consumer will be closed
+        return
```

Comment:
I followed existing behavior where a message would just be logged to standard error.
I can change the `checkErr` method profile to throw an `EOFException` or a custom subclass of `IOException` and catch that exception in the `run` method to output a message to standard error.
It might be better to close the `formatter` instance in a finally block there too as it is only closed on error in `checkErr` right now.

What do you think?